### PR TITLE
Integrate trace

### PR DESCRIPTION
**Pull request by shivamgcodes**
# Pull Request

## Description

Please include a summary of the changes and the related issue. Please also include relevant motivation and context.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Performance improvement
- [ ] Code refactoring
- [ ] Test improvements

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce.

- [ ] Unit tests
- [ ] Integration tests
- [ ] Manual testing
- [ ] Performance testing

## Test Configuration

* Python version:
* Operating System:
* Dependencies:

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

## Additional Notes

Add any other context about the pull request here.

## Summary by CodeRabbit

- New Features
  - Optional runtime tracing for model generation controllable via new NOVEUM_* and ENABLE_TRACING env vars.

- Documentation
  - Added Noveum-focused .env entries and minor .env comment clarification for GitHub key.
  - Added an OpenAI inference example demonstrating single/batch runs, parameter tuning, cost/token estimates, and connection checks.

- Chores
  - Added tracing, dotenv, and typing dependencies; dev typing config updated.

- Tests
  - Expanded tests covering tracing initialization, behavior, and edge cases.

### CODE REVIEW

File: src/novaeval/models/openai.py [hunk 1]
```diff
@@ -8,6 +8,7 @@
 import time
 from typing import Any, Optional, Union
 
+from noveum_trace import trace_llm
```
[the remark this answers is not in the document]
Comment (shivamgcodes) [reply src/novaeval/models/openai.py:11]: okay

File: src/novaeval/models/gemini.py [hunk 2]
```diff
@@ -16,6 +16,7 @@
 
 from google import genai
 from google.genai import types
+from noveum_trace import trace_llm
```
[the remark this answers is not in the document]
Comment (shivamgcodes) [reply src/novaeval/models/gemini.py:19]: done.

File: src/novaeval/models/base.py [hunk 3]
```diff
@@ -5,12 +5,39 @@
...
+
+def _trace_llm_noop(func: Optional[Callable] = None) -> Any:
+    # runtime no-op that behaves like the real decorator factory
+    if func is None:
+
+        def deco(f: Callable) -> Callable:
+            return f
+
+        return deco
+    return func
```
[the remark this answers is not in the document]
Comment (shivamgcodes) [reply src/novaeval/models/base.py:31]: thanks
